src/logger/appLogger.ts

```diff
@@ -14,9 +14,30 @@ export class AppLogger {
       fs.mkdirSync(this.logDirectory);
     }
   }
+ 
+  public static configureLogger(projectName?: string, shouldCreateLogFile: boolean = true) {
 
-  public static configureLogger() {
     this.createLogFolderIfNotExists();
+      const filepath = path.join(this.logDirectory,'.' );
+      console.log(filepath,"<<<filepath")
+      const oldFilePath =  path.join(this.logDirectory,`${projectName}-${new Date().toISOString().split('T')[0]}.log`);
+      console.log(oldFilePath,"<<<oldFilePath")
+    if(shouldCreateLogFile){
+      try {
+     
+        const files = fs.readdirSync(filepath);
+        const prefix = `${projectName}-${new Date().toISOString().split('T')[0]}`;
+        const count = files.filter(file => file.startsWith(prefix)).length;
+
+        if (count >= 1) {
+          const newPath = path.join(this.logDirectory, `${projectName}-${new Date().toISOString().split('T')[0]}-${count}.log`);
+          fs.renameSync(oldFilePath, newPath);
+        } 
+    } catch (err) {
+        console.error('Error:', err);
```

Comment:
this exception need to be handled.. not just logging to console but what happens after if we are unable to create log file

Reply:
The exception we are adding which his that whenever we are not able to generate the log file we will exit the process.